src/main/resources/templates/gardens/list.html

```diff
@@ -12,8 +12,50 @@
 <body>
 <div th:replace="fragments :: page-header"></div>
 <div class="container body-content">
+    <div>
+        <!-- Search form -->
+        <form th:action="@{/user/gardens/search}" method="GET">
+            <div style="margin-top: 5px">
+            <input class="form-control mr-sm-2"
+                   th:name="searchTerm"
+                   type="text"
+                   th:value="${searchTerm}"
+                   aria-label="Search">
+            </div>
+            <div class="form-group" style="margin-bottom: 0;">
+                <label th:for="features">Features: </label>
+                <th:block th:each="feature : ${features}">
+                    <input type="checkbox"
+                           th:name="featureIds"
+                           th:value="${feature.id}"
+                           th:if="${selectedFeatures.contains(feature)}"
+                           checked>
+                    <input type="checkbox"
+                           name="featureIds"
+                           th:value="${feature.id}"
+                           th:unless="${selectedFeatures.contains(feature)}">
+                    <label th:text="${feature.name} + '  '"></label>
+                </th:block>
+                <p>Note: all gardens are searched if no categories are selected.</p>
+            </div>
+            <div>
+            <button class="btn btn-outline-success btn-rounded btn-sm my-0" type="submit">Search</button>
+            </div>
+        </form>
+    </div>
+</div>
```

Comment:
This div closing tag can be deleted, as it prematurely closes the div tag on line 14. Without line 46, the tag is still closed on line 74.

Reply:
Good catch! Got it updated.